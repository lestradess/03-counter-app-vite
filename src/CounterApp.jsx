import { useState } from "react";
import PropTypes from "prop-types";

export const CounterApp = ( { value } ) =>
{
    const [counter, setCounter] = useState( value );

    const handleAdd = () => {
        //setCounter(counter + 1);
        setCounter( (c) => c + 1 )
        console.log( sdfalj )
    }
    const handleSubstract = () => setCounter( (c) => c - 1)
    const handleReset = () => setCounter( value )
    return (
        <>
            <h1>CounterApp</h1>
            <h2>{counter}</h2>

            <button onClick={handleAdd}> + </button>
            <button onClick={handleSubstract}> - </button>
            <button onClick={handleReset}> Reset </button>
        </>

    )
}
CounterApp.propTypes = {
    value: PropTypes.number,
}
CounterApp.defaultProps = {
    value: 0,
}