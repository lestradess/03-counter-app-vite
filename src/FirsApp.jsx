import PropTypes from "prop-types";



export const FirsApp = ({ title, subtitle }) => {



    return (
        <>
            <h2>{title}</h2>
            <h3>{subtitle}</h3>
            <p>By Lestrades</p>
        </>

    )
}
FirsApp.propTypes = {
    title: PropTypes.string.isRequired,
    subtitle: PropTypes.string.isRequired
}
FirsApp.defaultProps = {
    title: "No hay titulo",
    subtitle: "No hay",
}