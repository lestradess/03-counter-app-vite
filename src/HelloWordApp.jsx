export const HelloWordApp = () => {
  return (
    <h1>Hola mundo</h1>
  );
}
