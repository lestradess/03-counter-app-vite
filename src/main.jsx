import React from 'react';
import ReactDOM from 'react-dom/client';
import { FirsApp } from './FirsApp';
import { HelloWordApp } from './HelloWordApp';
import "./styles.css";




ReactDOM.createRoot(document.getElementById("root")).render(
    <React.StrictMode>
        <HelloWordApp />
        <FirsApp title="Empecemos a programar" subtitle="En React" />
    </React.StrictMode>
)