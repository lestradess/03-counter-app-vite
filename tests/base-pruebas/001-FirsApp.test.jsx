/** @jest-environment jsdom */

import { render } from "@testing-library/react"
import { FirsApp } from "../../src/FirsApp"

describe('Pruebas en FirsApp', () => {

    test("debe quedar con las propiedades por defecto", () => {
        const title = "Hola soy Goku";
        const { container } = render(<FirsApp title={ title } />);
        
        //Con esta funcion hace una instantanea de cómo esta la página
        expect(container).toMatchSnapshot();

    })
    test ("debe mostrar el título en un h1", () => {
        const title = "Hola soy Goku";
        const { container,getByText } = render(<FirsApp title={ title } />);

        expect(getByText(title)).toBeTruthy();        
    })
})