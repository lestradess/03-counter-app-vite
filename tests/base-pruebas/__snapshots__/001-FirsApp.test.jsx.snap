// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en FirsApp debe quedar con las propiedades por defecto 1`] = `
<div>
  <h2>
    Hola soy Goku
  </h2>
  <h3>
    No hay
  </h3>
  <p>
    By Lestrades
  </p>
</div>
`;
